extensions/pkg/webhook/utils_test.go

```diff
@@ -0,0 +1,682 @@
+				Expect(result).To(Equal(append(containers, newContainer)))
+			})
+
+			It("should replace the existing Container if it's not identical", func() {
+				existingContainer := corev1.Container{
+					Name:  "container1",
+					Image: "image3",
+				}
+				result := webhook.EnsureContainerWithName(containers, existingContainer)
+				Expect(result).To(Equal(
+					[]corev1.Container{
+						{
+							Name:  "container1",
+							Image: "image3",
+						},
+						{
+							Name:  "container2",
+							Image: "image2",
+						},
+					},
+				))
+			})
+
+			It("should do nothing to the existing Container if it's identical", func() {
+				identicalContainer := containers[0]
+				result := webhook.EnsureContainerWithName(containers, identicalContainer)
+				Expect(result).To(Equal(containers))
+			})
+		})
+
+		Describe("EnsureVPAContainerResourcePolicyWithName", func() {
+			var policies []vpaautoscalingv1.ContainerResourcePolicy
+			var modeAuto = vpaautoscalingv1.ContainerScalingModeAuto
+			var modeOff = vpaautoscalingv1.ContainerScalingModeOff
+
+			BeforeEach(func() {
+				policies = []vpaautoscalingv1.ContainerResourcePolicy{
+					{
+						ContainerName: "container1",
+						Mode:          &modeAuto,
+					},
+					{
+						ContainerName: "container2",
+						Mode:          &modeOff,
+					},
+				}
+			})
+
+			It("should add a new ContainerResourcePolicy if not present", func() {
+				newPolicy := vpaautoscalingv1.ContainerResourcePolicy{
+					ContainerName: "container3",
+					Mode:          &modeAuto,
+				}
+				result := webhook.EnsureVPAContainerResourcePolicyWithName(policies, newPolicy)
+				Expect(result).To(Equal(append(policies, newPolicy)))
+			})
+
+			It("should replace the existing ContainerResourcePolicy if it's not identical", func() {
+				existingPolicy := vpaautoscalingv1.ContainerResourcePolicy{
+					ContainerName: "container1",
+					Mode:          &modeOff,
+				}
+				result := webhook.EnsureVPAContainerResourcePolicyWithName(policies, existingPolicy)
+				Expect(result).To(Equal([]vpaautoscalingv1.ContainerResourcePolicy{
+					{
+						ContainerName: "container1",
+						Mode:          &modeOff,
+					},
+					{
+						ContainerName: "container2",
+						Mode:          &modeOff,
+					},
+				}))
+			})
+
+			It("should do nothing to the existing ContainerResourcePolicy if it's identical", func() {
+				identicalPolicy := policies[0]
+				result := webhook.EnsureVPAContainerResourcePolicyWithName(policies, identicalPolicy)
+				Expect(result).To(Equal(policies))
+			})
+		})
+
+		Describe("EnsurePVCWithName", func() {
+			var pvcs []corev1.PersistentVolumeClaim
+
+			BeforeEach(func() {
+				pvcs = []corev1.PersistentVolumeClaim{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pvc1",
+							Namespace: "namespace1",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "pvc2",
+							Namespace: "namespace2",
+						},
+					},
+				}
```

Comment:
A more real-world test data would be:
```suggestion
				pvcs = []corev1.PersistentVolumeClaim{
					{
						ObjectMeta: metav1.ObjectMeta{
							Name: "pvc1",
						},
						Spec: corev1.PersistentVolumeClaimSpec{
							Resources: corev1.ResourceRequirements{
								Requests: map[corev1.ResourceName]resource.Quantity{
									"storage": resource.MustParse("10Gi"),
								},
							},
						},
					},
					{
						ObjectMeta: metav1.ObjectMeta{
							Name: "pvc2",
						},
						Spec: corev1.PersistentVolumeClaimSpec{
							Resources: corev1.ResourceRequirements{
								Requests: map[corev1.ResourceName]resource.Quantity{
									"storage": resource.MustParse("20Gi"),
								},
							},
						},
					},
				}
```

Right now the `should replace the existing PersistentVolumeClaim if it's not identical` changes the namespace of a PVC which is not that real-world example as namespace of a obj is immutable.